### PR TITLE
Partial configuration

### PR DESCRIPTION
**Pull request by robinnedergard**
## Change
1. Simplified api-key handling, removed keyinfo and updated verifying/unpacking functionality to match recent authly changes.
2. Made properties of key optional.

## Rationale
1. Less confusing code.
2. Enables creation of key with partial key properties.

## Impact
1. Same functionality, will require some additional checks that key properties are actually set.
2. Enables creation of partially configured agent keys for easier creation of merchant keys.

## Risk
This should have no increased risks on the system. No extra risk analysis is necessary.

## Rollback
For rollback, it is enough to revert the commit and redeploy.

### CODE REVIEW

File: Merchant/UiKey.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import * as authly from "authly"
+import { Creatable as CardCreatable } from "./Card/Creatable"
+import { Card } from "./Card"
+import { Audience as KeyAudience } from "./Key/Audience"
+import { Key } from "./Key/index"
+import * as V1 from "./V1"
+
+export type UiKey = Omit<Key, "card"> & { card: Omit<Card, "acquirer" | "emv3d"> }
```
Comment (simonmika) [inline Merchant/UiKey.ts:8]: What does `UiKey` mean and why do we need it?

**Review by simonmika — CHANGES_REQUESTED** [review]

File: Merchant/UiKey.ts (hunk 1 above)
Comment (robinnedergard) [reply Merchant/UiKey.ts:8]: We use it in card-tokenizer

Comment (simonmika) [reply Merchant/UiKey.ts:8]: If you mean UI it is an abbreviation and up too the length of two letters we capitalize them both. However We should at least be able to use a better name. It has itself nothing to do with the UI. Maybe `Raw` would be the name?

But the question is why we can't use a Creatable in that context?

Comment (robinnedergard) [reply Merchant/UiKey.ts]: Replaced UiKey with function for extracting url, will be enough for card-tokeniser

File: Merchant/Key/index.spec.ts [hunk 2]
```diff
@@ -26,6 +26,28 @@ describe("Key", () => {
...
+				private:
+					"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjcwODIiLCJpYXQiOjE1ODM1MDM3MzA5NzAsImF1ZCI6InByaXZhdGUiLCJzdWIiOiJ0ZXN0IiwiYWdlbnQiOiJ0ZXN0IiwidHlwZSI6InRlc3QiLCJpZCI6InRlc3QiLCJuYW1lIjoiVGVzdCBBQiIsInVybCI6Imh0dHA6Ly9leGFtcGxlLmNvbSIsImRlc2NyaXB0b3IiOiJ0ZXN0IHRyYW5zYWN0aW9uIiwiY291bnRyeSI6IlNFIiwiYWNxdWlyZXIiOiJKdVpYd2swY2xzcnBMNGg1TDhqcTVsLThzWDFGZXdMWnBsalNKNlJITzcwWGNYU2d0MXdCenRHUlJoUTF3WW8zRUpTZ2J6Q0ZhTFkwVWpmaDF1aGNMRzBpTENKclpYa2lPaUl6Tm1VM05HRTJPUzAzWm1WbExUUmhNemN0WW1Oa09TMDJZVEUwTWpJd01qaG1aak1pTENKaWFXNGlPbnNpZG1sellTSTZJalF6T0RNd09TSXNJbTFoYzNSbGNtTmhjbVFpT2lJMU1qWTFOekVpZlgwIiwibWlkIjoiMTIzNCIsIm1jYyI6IjEyMzQiLCJlbXYzZCI6IlFQbDlaczdtemRDMUlBLVpLd1BQYnRSOS01VjlfYUJYcTR2VU5jN1hnbEtLOTlMQ0I2U0FneUhvcWFTU1I2VG56X2d4MVpKNmM4X2tHT1VNUFRYcnJqb2libTh0YTJWNUluMCJ9.U4Hpfcupl7tgjk-IVaYCemzuNsG15EZY1IxNgEdt02s",
+			},
+		},
```
Comment (simonmika) [inline Merchant/Key/index.spec.ts:49]: What type of keys are these? Please add comments with which type of secret they are signed.

**Review by simonmika — CHANGES_REQUESTED** [review]

File: Merchant/Key/index.spec.ts (hunk 2 above)
Comment (robinnedergard) [reply Merchant/Key/index.spec.ts]: Added signing information for example keys.

**Review by simonmika — APPROVED** [review]

**Pull request merged**